From baf4a55d4f4c7351897912f70148a14d5e587608 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Mon, 12 May 2014 20:27:29 +0200
Subject: [PATCH] getCameraID: return -1 on error instead of 0

0 is a valid camera ID, so don't reuse that as error value.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 src/comm/mavconn-imagestreamer.cc          | 2 +-
 src/interface/shared_mem/SHMImageClient.cc | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/comm/mavconn-imagestreamer.cc b/src/comm/mavconn-imagestreamer.cc
index bab487f..a901779 100644
--- a/src/comm/mavconn-imagestreamer.cc
+++ b/src/comm/mavconn-imagestreamer.cc
@@ -96,7 +96,7 @@ static void image_handler (const lcm_recv_buf_t *rbuf, const char * channel, con
 	// Check if there are images
 	uint64_t camId = client.getCameraID(msg);
 
-	if (camId != 0)
+	if (camId >= 0)
 	{
 		// Copy one image from shared buffer
 		if (!client.readStereoImage(msg, img, img2))
diff --git a/src/interface/shared_mem/SHMImageClient.cc b/src/interface/shared_mem/SHMImageClient.cc
index 7267d7b..2a446ec 100644
--- a/src/interface/shared_mem/SHMImageClient.cc
+++ b/src/interface/shared_mem/SHMImageClient.cc
@@ -126,7 +126,7 @@ SHMImageClient::getCameraID(const mavlink_message_t* msg)
 	if (msg->msgid != MAVLINK_MSG_ID_IMAGE_AVAILABLE)
 	{
 		// Instantly return if MAVLink message did not contain an image
-		return 0;
+		return -1;
 	}
 	else
 	{
-- 
1.9.1

